pilot/pkg/xds/proxy_dependencies.go

```diff
@@ -65,6 +66,17 @@ func checkProxyDependencies(proxy *model.Proxy, config model.ConfigKey) bool {
 		} else if proxy.PrevSidecarScope != nil && proxy.PrevSidecarScope.DependsOnConfig(config) {
 			return true
 		}
+	case model.Router:
+		if config.Kind == gvk.ServiceEntry {
+			// If config is ServiceEntry, name of the config is service's FQDN
+			svc, exist := push.ServiceIndex.HostnameAndNamespace[host.Name(config.Name)][config.Namespace]
+			if exist {
+				if !push.IsServiceVisible(svc, proxy.Metadata.Namespace) {
```

Comment:
A bit worried linking `IsServiceVisible` to this when its not actually the same function that is used to determine Services available to a gtw. It very likely functions correctly, but I am not super comfortable with these two decouple functions trying to compute the same thing with different logic.

Let me think on it... @hzxuzhonghu cc if you have ideas

Reply:
The check here is a superset of `GatewayServices`,  it's safe I think, maybe not that graceful.

@howardjohn As for gateway, we have generated the SidecarScope for it, I think we can use the same logic as SidecarProxy here, though it is still a superset of `GatewayServices`, but not that much